aep/general/0151/aep.md.j2

```diff
@@ -7,41 +7,89 @@ return some kind of promise to the user, and allow the user to check back in
+not** return a response with the result of the operation if the operation
+completes quickly.
 
-**Note:** User expectations can vary on what is considered "a significant
-amount of time" depending on what work is being done. A good rule of thumb is
-10 seconds.
+### Interface Definitions
 
-### Operation representation
+{% tab proto %}
 
-The response to a long-running request **must** be an [`Operation`][Operation].
+When using protocol buffers, the common component
+[`aep.api.Operation`][aep.api.Operation] is used.
 
-{% tab proto %}
+{% sample 'lro.proto', 'rpc WriteBook' %}
 
-Protocol buffer APIs **must** use the common component
-[`aep.api.Operation`][aep.api.Operation].
+- The response type **must** be `aep.api.Operation`. The `Operation` proto
+  definition **should not** be copied into individual APIs; prefer to use a
+  single copy (in monorepo code bases), or remote dependencies via a tool like
+  [Buf][buf.build].
+  - The response **must not** be a streaming response.
+- The method **must** include a [`aep.api.operation_info`][lro] annotation,
+  which **must** define both response and metadata types.
+  - The response and metadata types **must** be defined in the file where the
+    RPC appears, or a file imported by that file.
+  - If the response and metadata types are defined in another package, the
+    fully-qualified message name **must** be used.
+  - The response type **should not** be `google.protobuf.Empty` (except for
+    [`Delete`][aep-135] methods), unless it is certain that response data will
+    _never_ be needed. If response data might be added in the future, define an
+    empty message for the RPC response and use that.
+  - The metadata type is used to provide information such as progress, partial
+    failures, and similar information on each `GetOperation` call. The metadata
+    type **should not** be `google.protobuf.Empty`, unless it is certain that
+    metadata will _never_ be needed. If metadata might be added in the future,
+    define an empty message for the RPC metadata and use that.
+- APIs with messages that return `Operation` **must** implement the
+  `GetOperation` method of the [`Operations`][lro] service, and **may**
+  implement the other methods defined in that service. Individual APIs **must
+  not** define their own interfaces for long-running operations to avoid
+  inconsistency.
 
 {% tab oas %}
 
-OpenAPI services **must** use this [`JSON Schema Operation`][JSON Schema
-Operation] schema.
+{% sample 'lro.oas.yaml', 'paths' %}
+
+- `202` **must** be the only success status code defined.
+- The `202` response **must** define an `application/json` response body and no
+  other response content types.
+- The response body schema **must** be an object with `path`, `done`, `error`,
+  and `response` properties as described above for an `Operation`.
+- The response body schema **may** contain an object property named `metadata`
```

Comment:
**question:** Should that schema be included in this schema using `$ref` to ensure the responses can be validated? Should it be referenced somehow like it is in the protobuf option to allow re-usability of the Operation? Something else entirely?

Reply:
Same question for the response schema.

Reply:
hm.. maybe @rofrankel might have an opinion here? or @rambleraptor who's been working on a couple clients. Or @wora who's worked on AIPs.

It does feel like we should be describing the schema where possible. Protobuf schemas suffer a little bit from lack of dynamicism. Ideally we'd probably have individual LRO objects for each operation, but I imagine that being cumbersome is why aip.dev didn't recommend that.

The thing about LROs (to me) is that it's generally just a pattern to help combat disconnects. If a generated / dynamic client sees that the response is an LRO, all it would do is switch from extracting the response from the original request, to polling and extracting the response from the result of the LRO. Any metadata attached would generally be unused. 

If you want clients to use the information, you'd be better off adding it to the response of the original request, not metadata.